ocw/lib/influx.py

```diff
@@ -0,0 +1,45 @@
+import os
+import logging
+from influxdb_client import InfluxDBClient, Point
+from influxdb_client.client.write_api import SYNCHRONOUS, WriteApi
+from influxdb_client.client.exceptions import InfluxDBError
+from urllib3.exceptions import HTTPError, TimeoutError
+
+
+from webui.PCWConfig import PCWConfig
+
+logger = logging.getLogger(__name__)
+
+
+class Influx:
+    __client: WriteApi | None = None
+    VMS_QUANTITY: str = "vms_quantity"
+    IMAGES_QUANTITY: str = "images_quantity"
+    DISK_QUANTITY: str = "disk_quantity"
+
+    def __init__(self) -> None:
+        self.bucket: str = str(PCWConfig.get_feature_property("influxdb", "bucket"))
+        self.org: str = str(PCWConfig.get_feature_property("influxdb", "org"))
+
+    # this is implementation of Singleton pattern
```

Comment:
can you elaborate ? I don't understand

Reply:
> can you elaborate ? I don't understand

The code in `__new__` should contain only what is needed for implementing the Singleton pattern, leaving the rest of initialization to `__init__`

Reply:
I still can not figure out what can be moved to `__init__` :man_shrugging: . 
if we filter out "what needed for implementing the Singleton pattern" we will stay with this piece : 

```
 if os.getenv("INFLUX_TOKEN") is None:
                logger.warning("INFLUX_TOKEN is not set, InfluxDB will not be used")
            elif PCWConfig.has("influxdb/url"):
                url: str = str(PCWConfig.get_feature_property("influxdb", "url"))
                cls.__client = InfluxDBClient(
                    url=url,
                    token=os.getenv("INFLUX_TOKEN"),
                    org=str(PCWConfig.get_feature_property("influxdb", "org")),
                ).write_api(write_options=SYNCHRONOUS)
```

and whole point why I am using singleton to make sure that THIS code will run only ONE time . And first of all I can not see how to break it and even if there is a way whatever you will move to `__init__` will be executed on every object creation so than we loosing whole point of Singleton :man_shrugging:

Reply:
We can insert that block inside `if self.__client is None:` in `__init__`

Reply:
1. `self.__client` will be ALWAYS NONE in __init__ because self is pointer to object which was just created so this way are moving from having single InfluxDB instance to have InfluxDB instance in **every** object
2.  What the point to have Singleton than if __new__ contains only : 

```
if not hasattr(cls, "instance") or cls.instance is None:
            cls.instance = super(Influx, cls).__new__(cls)
        return cls.instance
```